mapping/field_dv.go

```diff
@@ -0,0 +1,203 @@
+// +build densevector
+
+package mapping
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"github.com/blevesearch/bleve/v2/geo"
+	"github.com/blevesearch/bleve/v2/document"
+)
+
+// A FieldMapping describes how a specific item
+// should be put into the index.
+type FieldMapping struct {
+	Name string `json:"name,omitempty"`
+	Type string `json:"type,omitempty"`
+
+	// Analyzer specifies the name of the analyzer to use for this field.  If
+	// Analyzer is empty, traverse the DocumentMapping tree toward the root and
+	// pick the first non-empty DefaultAnalyzer found. If there is none, use
+	// the IndexMapping.DefaultAnalyzer.
+	Analyzer string `json:"analyzer,omitempty"`
+
+	// Store indicates whether to store field values in the index. Stored
+	// values can be retrieved from search results using SearchRequest.Fields.
+	Store bool `json:"store,omitempty"`
+	Index bool `json:"index,omitempty"`
+
+	// IncludeTermVectors, if true, makes terms occurrences to be recorded for
+	// this field. It includes the term position within the terms sequence and
+	// the term offsets in the source document field. Term vectors are required
+	// to perform phrase queries or terms highlighting in source documents.
+	IncludeTermVectors bool   `json:"include_term_vectors,omitempty"`
+	IncludeInAll       bool   `json:"include_in_all,omitempty"`
+	DateFormat         string `json:"date_format,omitempty"`
+
+	// DocValues, if true makes the index uninverting possible for this field
+	// It is useful for faceting and sorting queries.
+	DocValues bool `json:"docvalues,omitempty"`
+
+	// SkipFreqNorm, if true, avoids the indexing of frequency and norm values
+	// of the tokens for this field. This option would be useful for saving
+	// the processing of freq/norm details when the default score based relevancy
+	// isn't needed.
+	SkipFreqNorm bool `json:"skip_freq_norm,omitempty"`
+
+	// Dimension of the dense vector
+	Dims int `json:"dims,omitempty"`
+
+	// Similarity is the similarity algorithm used for scoring
+	// dense_vector fields. See also SimilarityDefault and SimilarityValid
+	Similarity string `json:"similarity,omitempty"`
+}
+
+// Default Similarity value for dense_vector fields
+const SimilarityDefault = "l2_norm"
+
+// Set of Valid Similarity values for dense_vector fields
+var SimilarityValid = map[string]bool{
+	"l2_norm":     true, // default
+	"cosine":      false,
+	"dot_product": false,
+}
+
+func NewDenseVectorFieldMapping() *FieldMapping {
+	return &FieldMapping{
+		Type:         "densevector",
+		Store:        false,
+		Index:        true,
+		IncludeInAll: false,
+		DocValues:    false,
+	}
+}
+
+func (fm *FieldMapping) processDenseVector(propertyMightBeDenseVector interface{},
+	pathString string, path []string, indexes []uint64, context *walkContext) {
+	propertyVal := reflect.ValueOf(propertyMightBeDenseVector)
+	if !propertyVal.IsValid() {
+		return
+	}
+
+	if propertyVal.Kind() == reflect.Slice {
+		dv := make([]float32, propertyVal.Len())
```

Comment:
Above this line, we need to validate , whether len of vector conform the dimensions set at the time of index creation.

This is important for zapx to determine Dimension and Similarity value of a FAISS Index to be created from a Batch of Documents (with dense vector as a field)